### PR TITLE
Rpc example and documentation

### PR DESCRIPTION
**Pull request by gridbugs**
This adds a simple example program that runs an RPC client and updates the RPC documentation to reference the example, give a little bit more detail, and mention `dune-rpc-lwt`.

### CODE REVIEW

File: doc/rpc.rst [hunk 1]
```diff
@@ -31,15 +31,22 @@ will work with all servers ``>= X``. However, it will not work with any servers
 ``< X``.
 
 The library contains a client implementation parameterized over a concurrency
```
Comment (christinerose) [inline doc/rpc.rst:33]:
```suggestion
The library contains a client implementation parameterised over a concurrency
```

"s" in British spelling, unless this is a specific CS term that must be spelled that way.

File: doc/rpc.rst [hunk 2]
```diff
@@ -31,15 +31,22 @@ will work with all servers ``>= X``. However, it will not work with any servers
...
+monad. The ``dune-rpc-lwt`` package provides an instantiation of the RPC client
+defined in ``dune-rpc`` using lwt_ to implement the concurrency monad.
+It is of course possible to instantiate the dune RPC client using alternative
+concurrency implementations.
 
-The library provides an API to do the following:
+
+The ``dune-rpc`` library provides an API to do the following:
 
 - Initialize an RPC session over a channel
```
Comment (christinerose) [inline doc/rpc.rst:42]:
```suggestion
- Initialise an RPC session over a channel
```

File: doc/rpc.rst [hunk 3]
```diff
@@ -31,15 +31,22 @@ will work with all servers ``>= X``. However, it will not work with any servers
...
 - Send RPC requests and notifications
 - Handle notifications received from the server
 - Definitions of available requests, notifications, and their associated types.
+- Subscribe to streams of build progress events
+
+`Here <rpc_client_example_>`_ is an example of a simple dune RPC client which
```
Comment (christinerose) [inline doc/rpc.rst:48]:
```suggestion
`Here <rpc_client_example_>`_ is an example of a simple Dune RPC client that
```

File: example/rpc_client.t/README.org [hunk 4]
```diff
@@ -0,0 +1,13 @@
+This project contains an executable =rpc_client= which connects to the RPC
+server started when dune is run in watch mode. To use this program, start dune
```
Comment (christinerose) [inline example/rpc_client.t/README.org:2]:
```suggestion
server started when Dune is run in watch mode. To use this program, start Dune
```

File: example/rpc_client.t/run.t [hunk 5]
```diff
@@ -0,0 +1,15 @@
+Start dune in watch mode, sending errors to /dev/null to surpress the alerts
```
Comment (christinerose) [inline example/rpc_client.t/run.t:1]:
```suggestion
Start Dune in watch mode, sending errors to `/dev/null` to suppress the alerts
```

File: example/rpc_client.t/run.t [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+  $ while ! test -f _build/default/rpc_client.exe; do sleep 1; done
+
+Run the program. The program takes care of shutting down the build server.
+  $ _build/default/rpc_client.exe
+  Sending ping to server...
+  Got response from server...
+  Creating progress stream...
+  Waiting for next progress event...
+  Got progress_event: Success
+  Shutting down RPC server...
```
Comment (christinerose) [inline example/rpc_client.t/run.t:15]: Should these lines of code be wrapped in a code block?

**Review by christinerose — CHANGES_REQUESTED** [review]

File: example/rpc_client.t/run.t (hunk 6 above)
Comment (gridbugs) [reply example/rpc_client.t/run.t:15]: What do you mean by code block? Just clarifying that this file defines a test; it's not just documentation.

File: example/rpc_client.t/README.org [hunk 7]
```diff
@@ -0,0 +1,13 @@
+This project contains an executable =rpc_client= which connects to the RPC
```
[the remark this answers is not in the document]
Comment (gridbugs) [reply example/rpc_client.t/README.org:1]: Yeh sure. Is there a way to add tests for examples?

File: example/rpc_client.t/run.t (hunk 6 above)
Comment (christinerose) [reply example/rpc_client.t/run.t:15]: I mean in monospace: 
```
$ _build/default/rpc_client.exe
  Sending ping to server...
  Got response from server...
  Creating progress stream...
  Waiting for next progress event...
  Got progress_event: Success
  Shutting down RPC server...
  ```

Comment (gridbugs) [reply example/rpc_client.t/run.t:15]: Are you assuming that this is a markdown file? This file is a cram test. It's a script that runs a test with inline comments, commands, and their expected output. It doesn't get rendered by github.

Comment (christinerose) [reply example/rpc_client.t/run.t:15]: OH! Thank you for that clarification! My mistake and oversight.

**Review by christinerose — APPROVED** [review]

File: example/rpc_client.t/README.org (hunk 7 above)
[the remark this answers is not in the document]
Comment (rgrinberg) [reply example/rpc_client.t/README.org:1]: Nothing specifically for example. Cram or mdx I suppose. I'd use cram to keep things simple.

[the remark this answers is not in the document]
Comment (gridbugs) [reply example/rpc_client.t/README.org]: Cool, moved the test to a cram test defined in otherlibs/dune-rpc-lwt

**Comment by gridbugs** [comment]
I moved the example inside otherlibs/dune-rpc-lwt and changed the README to markdown.

**Comment by gridbugs** [comment]
> @gridbugs could you run `make fmt`?

Done!

**Pull request merged**